aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -46,125 +45,120 @@ Tensor fbgemm_linear_int8_weight_fp32_activation(
   // scalars. But in this case, we're doing whole-tensor quantization so we just
   // pass a pointer to the scale values (and internally ReQuantizeFor Float
   // won't index past 0
-  float weight_scale_float = static_cast<float>(weight_scale.to<double>());
-  int32_t weight_zero_point_int32 =
+  const float weight_scale_float =
+      static_cast<float>(weight_scale.to<double>());
+  const int32_t weight_zero_point_int32 =
       static_cast<int32_t>(weight_zero_point.to<int64_t>());
 
-  auto bias_contig = bias.contiguous();
+  const auto& bias_contig = bias.is_contiguous() ? bias : bias.contiguous();
 
-  // The resulting matrix here is 2-D, let's view it with the original
-  // left hand dimensions of the input.
-  // Here are two examples:
-  // 1. If the input tensor is {M, K}, the output tensor is {M, N}.
-  // 2. If the input tensor is {b, M, K}, the output tensor is {b, M, N}.
-  std::vector<int64_t> out_sizes = input.sizes().vec();
-  out_sizes.back() = N;
   // Allocate output Tensor and a buffer for fbgemmPacked to use
-  auto output = at::zeros(
-      out_sizes, bias.options().dtype(at::kFloat));
-  auto buffer = at::zeros_like(output, output.options().dtype(at::kInt));
+  std::vector<int64_t> output_size = input.sizes().vec();
+  output_size.back() = N;
+  Tensor output = at::empty(output_size, input.options().dtype(at::kFloat));
+  Tensor buffer = at::empty(output_size, input.options().dtype(at::kInt));
 
   // Pull out the PackBMatrix instance from the owning tensor
-  auto& packB = cpp_custom_type_hack::cast<fbgemm::PackBMatrix<int8_t>>(packed);
+  auto& pack_b =
+      cpp_custom_type_hack::cast<fbgemm::PackBMatrix<int8_t>>(packed);
 
-  int num_tasks = at::get_num_threads();
+  const int num_tasks = at::get_num_threads();
   at::parallel_for(0, num_tasks, 1, [&](int64_t begin, int64_t end) {
     // This operation does the following:
-    // 1) Quantizes the input matrix given the statistics we've calculated above
+    // 1) Quantizes the input matrix given the statistics we've calculated
+    //    above.
     // 2) Creates a "row buffer" vector with offset values that must be added
-    //    to the integer matrix multiplication operation to ensure correctness
+    //    to the integer matrix multiplication operation to ensure correctness.
     // 3) Packs the resulting quantized matrix into vector-register and cache
     //    friendly tiles.
     //
     //  Note this is not executed eagerly, but rather within the fbgemmPacked
     //  call below.
-    fbgemm::PackAWithQuantRowOffset<uint8_t> packA(
-        /*trans=*/fbgemm::matrix_op_t::NoTranspose,
-        /*nRow=*/M,
-        /*nCol=*/K,
-        /*smat=*/input_ptr,
-        /*ld=*/K,
-        /*pmat=*/nullptr, // packA manages ownership of `pmat`
-        /*scale=*/q_params.scale,
-        /*zero_pt=*/q_params.zero_point);
+    fbgemm::PackAWithQuantRowOffset<uint8_t> pack_a(
+        /* trans = */ fbgemm::matrix_op_t::NoTranspose,
```

Comment:
Nit: the spaces added in the comment: Is it some CPP standard? Changing this might mess up the blame.

Reply:
Done.